ethtx_ce/frontend/deps.py

```diff
@@ -10,17 +10,62 @@
 #  See the License for the specific language governing permissions and
 #  limitations under the License.
 
+import os
+from typing import Tuple
+
+import pkg_resources
+from flask import current_app, Blueprint
 from flask_httpauth import HTTPBasicAuth
+from git import Repo
 
 from ethtx_ce.config import Config
 
+bp = Blueprint("deps", __name__)
 auth = HTTPBasicAuth()
 
 
+@bp.before_app_first_request
+def read_ethtx_versions() -> None:
+    """Read ethtx and ethtx_ce versions."""
+    ethtx_version = pkg_resources.get_distribution("ethtx").version
+
+    try:
+        remote_url, sha = _get_version_from_git()
+    except Exception:
+        remote_url, sha = _get_version_from_docker()
+    ethtx_ce_version = f"{remote_url}/tree/{sha}"
+
+    current_app.config["ethtx_version"] = ethtx_version
+    current_app.config["repo_version"] = ethtx_ce_version
+
+
 @auth.verify_password
 def verify_password(username: str, password: str) -> bool:
     """Verify user, return bool."""
     return (
         username == Config.ETHTX_ADMIN_USERNAME
         and password == Config.ETHTX_ADMIN_PASSWORD
     )
+
+
+def _get_version_from_git() -> Tuple[str, str]:
+    """Get EthTx CE version from .git"""
+    root = os.path.dirname(os.path.dirname(os.path.dirname(__file__)))
+    repo = Repo(root, search_parent_directories=True)
+
+    remote_url = repo.remote("origin").url.replace(".git", "")
+    sha = repo.head.object.hexsha
+
+    return remote_url, sha
+
+
+def _get_version_from_docker() -> Tuple[str, str]:
+    """Get EthTx CE version from file."""
+    try:
+        with open("/app/get_version") as f:
+            url_sha = f.readline().split(",")
+    except OSError:
+        url_sha = ["", ""]
```

Comment:
Use `os.path.isfile` to check if file exists, instead of this `try except` block